csharp/src/Ice/RemoteException.cs

```diff
@@ -19,22 +19,35 @@ public class RemoteException : System.Exception
         /// in a remote server.</summary>
         public bool ConvertToUnhandled { get; set; } = false;
 
+        /// <summary>When true, a custom message was provided to the constructor. Otherwise, false.</summary>
+        protected bool HasCustomMessage { get; } = false;
+
```

Comment:
Shouldn't we support a protected DefaultMessage virtual method at this level rather than this boolean property? If non-empty, the remote exception default constructor would provide this default message to the System.Exception message constructor.

Reply:
I think it's cleaner that way. Keep in mind that the system provides a default internationalized message: the default system-provided message (when you don't pass a message at all) is _not_ empty.

Reply:
Yes, the default system message isn't empty and my suggestion is not to always override it. We'd only override it if DefaultMessage is set. Just like it is for request failed exceptions, this could be useful for remote exceptions in general.

Reply:
When the application provides a custom message explicitly - or the unmarshaling code sets the message - we want this explicit message to prevail. So you're suggesting a combination of HasCustomMessage plus a DefaultMessage that is null by default (null meaning use the default system message)?

Reply:
I was thinking of getting the `DefaultMessage` from the` RemoteException` constructors that don't take a message but calling a virtual method from the constructor isn't a good idea! So yes this would still require a private boolean in `RemoteException` if we want to provide this at the `RemoteException` level... something like this:
```
private bool _hasCustomMessage = false;
protected virtual string DefaultMessage => null;
public override string Message => _hasCustomMessage ? base.Message : (DefaultMessage ?? base.Message);
```

Reply:
Fixed along those lines.